### ISSUE TITLE
Add more machine learning algorithms

### ISSUE THREAD

**Issue by param087** [param087/swiftML#42]
**Algorithms**

* Linear Model
  - [x] Ordinary Least Squared Linear Regression
  - [x]  Gradient Descent Linear Regression
  - [x]  Stochastic Gradient Descent Linear Regression
  - [x]  Logistic Regression 
  - [ ]  Lasso Regression
  - [ ]  Ridge Regression
  - [ ]  Elastic Regression
  - [ ]  Bayesian Regression
  - [ ]  SVD Regression

* SVM (Support Vector Machine)
  - [ ] SVC
  - [ ] SVR
  - [ ] LinearSVC
  - [ ] LinearSVR
  - [ ] NuSVC
  - [ ] NuSVR

* Cluster
  - [x] KMeans
  - [ ] DBSCAN
  - [ ] Spectral clustering
  - [ ] Hierarchical clustering

* Decomposition
  - [x] PCA
  - [ ] Sparse PCA
  - [ ] Kernel PCA
  - [ ] Incremental PCA

* Naive Bayes
  - [x] Bernoulli Naive Bayes
  - [x] Nultinomial Naive Bayes
  - [x] Gaussian Naive Bayes
  - [ ] Complement Naive Bayes

* Neighbors
  - [x] Nearest Neighbors Classifier
  - [x] Nearest Neighbors Regressor
  - [ ] Nearest centroid classifier

* Tree
  - [x] Decision Tree

* Ensemble 
    - [x] Random Forest
    - [ ] Gradient Boosting
    - [ ] XGBoost
    - [x] AdaBoost
    - [ ] Stacking
    - [ ] Voating

Some of the Tree base algorithms are implemented but need some review.

**Comment by Lord-V15** [comment]
I could help you complete the "Nearest centroid classifier". I'd love to help out the community driven project.

**Comment by param087** [comment]
> I could help you complete the "Nearest centroid classifier". I'd love to help out the community driven project.

Go ahead but do follow https://github.com/param087/swiftML/blob/master/CONTRIBUTING.md

**Comment by das-ankur** [comment]
I want to implement linear models. Can I start working?

**Comment by param087** [comment]
Yes @das-ankur, you can start working on it. If you need any help, please reach out to me.
